cdc/kv/shared_stream.go

```diff
@@ -125,16 +140,7 @@ func newRequestedStream(streamID uint64) *requestedStream {
 	return stream
 }
 
-func (s *requestedStream) run(ctx context.Context, c *SharedClient, rs *requestedStore) (canceled bool) {
-	isCanceled := func() bool {
-		select {
-		case <-ctx.Done():
-			return true
-		default:
-			return false
-		}
-	}
-
+func (s *requestedStream) run(ctx context.Context, c *SharedClient, rs *requestedStore) error {
 	if err := version.CheckStoreVersion(ctx, c.pd, rs.storeID); err != nil {
```

Comment:
Seems we just need to move `CheckStoreVersion` out from `run`.

Reply:
Change `requestedStream.run` from return `bool` to return `error` enable us to handle more kinds of error in the future.
And what's the benefits of move `CheckStoreVersion` out from run?